From 24c6faa5aaaec6d5408c0505c258ae8ace9d0828 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 12 Aug 2024 22:19:25 +0800
Subject: [PATCH 3/3] fix: do not tint terminal by default

Let's face it, it's weird and we have much better ways to notify the user
about a system running in a container. Change the default behavior (tint by
default) to only tint when the user specifies a --background= parameter.
---
 src/shared/pretty-print.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/pretty-print.c b/src/shared/pretty-print.c
index 4692a6aedc80..695c986c9e12 100644
--- a/src/shared/pretty-print.c
+++ b/src/shared/pretty-print.c
@@ -453,7 +453,7 @@ bool shall_tint_background(void) {
 
         cache = getenv_bool("SYSTEMD_TINT_BACKGROUND");
         if (cache == -ENXIO)
-                return (cache = true);
+                return (cache = false);
         if (cache < 0)
                 log_debug_errno(cache, "Failed to parse $SYSTEMD_TINT_BACKGROUND, leaving background tinting enabled: %m");
 
-- 
2.47.0

